example/remap/remap.cc

```diff
@@ -210,8 +212,8 @@ TSRemapDeleteInstance(void *ih)
   delete ri;
 }
 
-static volatile unsigned long processing_counter = 0; // sequential counter
-static int arg_index                             = 0;
+static std::atomic<uint64_t> processing_counter; // sequential counter
+static int arg_index;
 
```

Comment:
Hmm I'm not sure about changing unsigned long to uint64_t.  Are we sure unsigned long is not uint32_t on any system that TS runs on?

If you're on a 32-bit architecture, with no bus-locking 64-bit load/store instructions, I'm pretty sure using std::atomic<*uint64_t*> will generate code with retry loops.

Reply:
a) We don't support 32bit
b) This is an example plugin and that is just a counter for monotonically increasing values

TBH, I was sort of just being cute by making that change. There was a commetn that said "use a mutex" or similar, so I just added atomics.